Robot.java_03_14_17

```diff
@@ -1,418 +1 @@
-			//do I have to cast left as a goodDriveside instead of speedcontroller?
-		
-		if (state.equals("forward1")) {
-			moveForward(9.5167);
-			//10.83 feet = 129.96 inches
-			//9.5167 feet = 114.3 inches
-			//have to calculate again with the length of the robot and the distance from the center
-			//from left and right positions not the middle
-			state = "turn1";
-				
-		}
-		else if (state.equals("turn1")) {
-			changeAngle(120); 
-			state = "forward2";
-			
-		}
-		else if (state.equals("forward2")) {
-			moveForwardSlow(.5);
-			state = "stop";
-			
-		}
-		else if (state.equals("stop")){
-			left.set(0);
-			right.set(0);
-		}
-		else {
-			left.set(0);
-			right.set(0);
-			
-		}
-
-		
-		
-		
-	}
-	
-	
-	
-	@Override
-	public void teleopInit() {
-		// This makes sure that the autonomous stops running when
-		// teleop starts running. If you want the autonomous to
-		// continue until interrupted by another command, remove
-		// this line or comment it out.
-		System.out.println("Initiating Teleop Mode");
-
-		if (autonomousCommand != null)
-			autonomousCommand.cancel();
-		
-		driveTrain = new RobotDrive(leftSpark, rightSpark);
-
-		
-		leftEncoder.reset();
-		
-		
-		
-		
-	}
-
-	/**
-	 * This function is called periodically during operator control
-	 */
-	
-	@Override
-	public void teleopPeriodic() {
-		
-//needs a String/boolean parameter		
-		Scheduler.getInstance().run();
-		
-/*		if (cameraMode.equalsIgnoreCase("BACK")){
-			
-			cameraServer.getInstance().startAutomaticCapture();
-		}
-		
-		else if (cameraMode.equalsIgnoreCase("FRONT")){
-	//		cameraServer2.getInstance().startAutomaticCapture();
-			
-		} */
-
-		//tank drive - Aliya / Halie 
-		
-		
-		if (driveMode.equalsIgnoreCase("TANK")) {
-
-			//driveTrain.tankDrive(-1*stick.getY(),-1*stick.getRawAxis(5));
-			driveTrain.tankDrive(-1*stick2.getY(), stick.getY());
-
-		}
-		
-		//left axis: stick.getY();
-		//right axis: stick.getRawAxis(5);
-
-		else if(driveMode.equalsIgnoreCase("ARCADE")){
-		//arcade drive
-			driveTrain.arcadeDrive(-stick.getY(),stick.getX());
-			//driveTrain.arcadeDrive(stick);
-		}
-		
-		else if(driveMode.equalsIgnoreCase("BACK TANK")){
-			driveTrain.tankDrive(-1*stick.getY(), stick2.getY());
-			
-		}
-		
-		//winch control - Halie 
-		winch.tankDrive(stick.getRawAxis(3), 0);
-		
-
-	}
-			
-
-	/**
-	 * This function is called periodically during test mode
-	 */
-	@Override
-	public void testPeriodic() {
-		LiveWindow.run();
-	}
-	 
-	
-	
-	
-	public double feetToEncUnits(double feet) {
-		
-		return feet*12/(6*Math.PI)*360/2400;
-		
-		
-		
-	}
-	
-	
-	
-	public void moveForward(double feet) {
-		
-		if (leftEncoder.getDistance() < feetToEncUnits(feet)) {
-			//rotations: X * 2400 / 360 
-
-			//feet: 
-			left.set(.5);
-			right.set(.5);
-		}
-
-		
-		else {
-			left.set(0);
-			right.set(0);
-			
-		} 
-		
-	}
-	
-	public boolean changeAngle(double degrees) {
-		
-		boolean go = true;
-		while (go) {
-			if (gyro.getAngle() == degrees) {
```

Comment:
This is unlikely to turn the way you want it to. Because of inertia, the robot will probably never stop exactly at degrees, so it will go past that point and keep on going.

Also, you will only turn counter-clockwise in this code even if the target angle is 10 degrees clockwise.

You can make a few simple changes to this flow so that it homes in on the heading you want and stops there. Think about what we talked about with the PID controller where the P term can be used to bring it back to its target. 

(Note, you could actually do some refactoring and use the PIDController class to do this, but given the time constraints, I wouldn't recommend it. That's a bigger project)